### PR TITLE
Release - 2.1.0

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened
Bugs:
- #53
- #54
- #56
- #60

Chores:
- #52
- #55
- #59
- #61
 
## Insight
N/A 

## Proof Of Work
N/A

### CODE REVIEW

File: app/controllers/api/v1/passwords_controller.rb [hunk 1]
```diff
@@ -6,10 +6,14 @@ class PasswordsController < Devise::PasswordsController
       include API::V1::OauthApplicationVerifiable
 
       def create
```
Comment (team-nimblehq) [inline app/controllers/api/v1/passwords_controller.rb:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/435daeb7c60d599ae8c998be4074bc12fe37821b/app/controllers/api/v1/passwords_controller.rb#L8"></span>Remove unused methods (api::v1::passwordscontroller#create)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/registrations_controller.rb [hunk 2]
```diff
@@ -5,6 +5,8 @@ module V1
     class RegistrationsController < Devise::RegistrationsController
       include API::V1::OauthApplicationVerifiable
 
+      before_action :configure_permitted_parameters, if: :devise_controller?
+
       def create
```
Comment (team-nimblehq) [inline app/controllers/api/v1/registrations_controller.rb:10]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/435daeb7c60d599ae8c998be4074bc12fe37821b/app/controllers/api/v1/registrations_controller.rb#L10"></span>Remove unused methods (api::v1::registrationscontroller#create)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/graphql_controller.rb [hunk 3]
```diff
@@ -30,4 +30,25 @@ def ensure_hash(ambiguous_param)
       raise ArgumentError, "Unexpected parameter: #{ambiguous_param}"
     end
   end
+
+  # :reek:FeatureEnvy
+  # rubocop:disable Metrics/MethodLength
+  def doorkeeper_unauthorized_render_options(error: nil)
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:36]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/435daeb7c60d599ae8c998be4074bc12fe37821b/app/controllers/graphql_controller.rb#L36"></span>Remove unused methods (graphqlcontroller#doorkeeper_unauthorized_render_options)</td>
    </tr>
  </tbody>
</table>

File: app/models/survey.rb [hunk 4]
```diff
@@ -29,9 +29,7 @@ class Survey < ApplicationModel
 
   class << self
     def all
-      @all ||= begin
-        JSON.parse(File.read(Rails.root.join('lib', 'assets', 'surveys.json'))).map(&method(:new))
-      end
+      @all ||= JSON.parse(File.read(Rails.root.join('lib', 'assets', 'surveys.json'))).map(&method(:new))
     end
 
     def where(attributes = {})
```
Comment (team-nimblehq) [inline app/models/survey.rb:35]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/435daeb7c60d599ae8c998be4074bc12fe37821b/app/models/survey.rb#L35"></span>Remove unused methods (survey#where)</td>
    </tr>
  </tbody>
</table>

File: app/models/user.rb [hunk 5]
```diff
@@ -8,11 +8,19 @@ class User < ApplicationRecord
   has_many :access_tokens,
```
Comment (team-nimblehq) [inline app/models/user.rb:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/435daeb7c60d599ae8c998be4074bc12fe37821b/app/models/user.rb#L8"></span>Specify an <code>:inverse_of</code> option.</td>
    </tr>
  </tbody>
</table>

**Comment by team-nimblehq** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/435daeb7c60d599ae8c998be4074bc12fe37821b/app/controllers/graphql_controller.rb#L6">app/controllers/graphql_controller.rb#L6</a> - Remove unused methods (graphqlcontroller#create)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/435daeb7c60d599ae8c998be4074bc12fe37821b/app/models/survey.rb#L39">app/models/survey.rb#L39</a> - Remove unused methods (survey#find_by)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by junan — APPROVED** [review]

**Review by ankitkalia1195 — APPROVED** [review]

**Pull request merged**